lib/langchain/conversation.rb

```diff
@@ -59,7 +59,19 @@ def add_examples(examples)
     def message(message)
       human_message = HumanMessage.new(message)
       @memory.append_message(human_message)
-      ai_message = llm_response(human_message)
+      ai_message = llm_response
+      @memory.append_message(ai_message)
+      ai_message
+    end
+
+    # Message the model with the function name, content and return the response.
+    # @param message [String] The content or arguments for the function call
+    # @param name [String] The name of the function to be called
+    # @return [String] The response from the function call
+    def function(message, name)
+      function_message = FunctionMessage.new(message, {name: name})
+      @memory.append_message(function_message)
+      ai_message = llm_response
```

Comment:
so, is the idea that every time we invoke a function, we then invoke the LLM to convert the result to a more friendly result?

I was thinking that it might not be always necessary and instead we need a more dumb API that just lets us add the result of having called the function that was suggested by the LLM, something like https://github.com/andreibondarev/langchainrb/pull/286/files#diff-9672ba0f87466bc23c7508052b5e8f37280c2ec8efd46faecacf4c9d72f6c24dR56-R61 WDYT?

Reply:
Why would it not be necessary? This is a real necessity of an MVP product. The flow goes like this:

user: I want to make an appointment

Here it gets interesting. Sometimes OpenAI functions call the function with empty or nonsense arguments. Then with this method is possible to "double-check":

``` ruby
chat.function("The appointment time is empty", "appointment")
```
#### Get the info like time, day, and so on
ai:  call function appointment with arguments
#### Everything is valid:
``` ruby
chat.function("Confirm the appointment", "appointment")
```
#### The function process and give the result:

``` ruby
chat.function("Appointment confirmed", "appointment")
```

It could be solved with different approaches, but it is just an example.

So, if we just append using that method and don't call the LLM to process the result, doesn't make sense.